suite-native/module-trading/src/hooks/useTradingBuyForm.ts

```diff
@@ -197,22 +207,49 @@ const useQuoteQuoteIdChangeEffect = (form: TradingBuyForm) => {
     }, [selectedQuote, isAmountInSats, symbol, form]);
 };
 
+const useValidations = (form: TradingBuyForm, limits: AmountLimitProps | undefined) => {
+    const { translate } = useTranslate();
+    const quotes = useSelector(selectValidTradingBuyQuotes);
+    const quoteRequest = useSelector(selectTradingBuyQuotesRequest);
+    const prevLimits = useRef<AmountLimitProps | undefined>(undefined);
+
+    if (prevLimits.current !== limits) {
+        prevLimits.current = limits;
+        form.trigger(['cryptoValue', 'fiatValue']);
+    }
+
+    useEffect(() => {
```

Comment:
If I have it right, this `useEffect` can be simplified to:

```suggestion
    useEffect(() => {
        if (!quoteRequest || quotes.length > 0 || limits) {
            form.setValue('generalAlert', undefined);

            return;
        }

        form.setValue('generalAlert', translate('moduleTrading.validators.noQuotes'));
    }, [quoteRequest, quotes, limits, form, translate]);
```

wdyt?

Reply:
Updated. Calling `form.setValue` always triggers re-render of form, even if value is same as before 🤷 . Therefore refactored to rewrite value only when necessary.